// @jayess/raylib/index.js
import { setTraceLogLevelNative, setConfigFlagsNative, initWindowNative, closeWindowNative } from "./native/raylib.bind.js";
import { isWindowReadyNative, windowShouldCloseNative, setWindowTitleNative, setWindowSizeNative, getScreenWidthNative, getScreenHeightNative } from "./native/raylib.bind.js";
import { beginDrawingNative, endDrawingNative, clearBackgroundNative, drawCircleNative, drawTextNative } from "./native/raylib.bind.js";
import { genImageColorNative, loadImageNative, loadImageFromBytesNative, unloadImageNative, isImageReadyNative, getImageWidthNative, getImageHeightNative } from "./native/raylib.bind.js";
import { loadTextureNative, loadTextureFromImageNative, unloadTextureNative, isTextureReadyNative, getTextureWidthNative, getTextureHeightNative, drawTextureNative } from "./native/raylib.bind.js";
import { isKeyPressedNative, isKeyDownNative, isMouseButtonDownNative, getMouseXNative, getMouseYNative, getMousePositionNative } from "./native/raylib.bind.js";
import { isGamepadAvailableNative, getGamepadAxisCountNative, isGamepadButtonDownNative, getGamepadNameNative } from "./native/raylib.bind.js";
import { setTargetFPSNative, getFrameTimeNative, getTimeNative, setRandomSeedNative, getRandomValueNative } from "./native/raylib.bind.js";

export function setTraceLogLevel(level) {
  return setTraceLogLevelNative(level);
}

export function setConfigFlags(flags) {
  return setConfigFlagsNative(flags);
}

export function initWindow(width, height, title) {
  return initWindowNative(width, height, title);
}

export function closeWindow() {
  return closeWindowNative();
}

export function isWindowReady() {
  return isWindowReadyNative();
}

export function windowShouldClose() {
  return windowShouldCloseNative();
}

export function setWindowTitle(title) {
  return setWindowTitleNative(title);
}

export function setWindowSize(width, height) {
  return setWindowSizeNative(width, height);
}

export function getScreenWidth() {
  return getScreenWidthNative();
}

export function getScreenHeight() {
  return getScreenHeightNative();
}

export function beginDrawing() {
  return beginDrawingNative();
}

export function endDrawing() {
  return endDrawingNative();
}

export function clearBackground(color) {
  return clearBackgroundNative(color);
}

export function drawCircle(x, y, radius, color) {
  return drawCircleNative(x, y, radius, color);
}

export function drawText(text, x, y, size, color) {
  return drawTextNative(text, x, y, size, color);
}

export function genImageColor(width, height, color) {
  return genImageColorNative(width, height, color);
}

export function loadImage(path) {
  return loadImageNative(path);
}

export function loadImageFromBytes(type, bytes) {
  return loadImageFromBytesNative(type, bytes);
}

export function unloadImage(image) {
  return unloadImageNative(image);
}

export function isImageReady(image) {
  return isImageReadyNative(image);
}

export function getImageWidth(image) {
  return getImageWidthNative(image);
}

export function getImageHeight(image) {
  return getImageHeightNative(image);
}

export function loadTexture(path) {
  return loadTextureNative(path);
}

export function loadTextureFromImage(image) {
  return loadTextureFromImageNative(image);
}

export function unloadTexture(texture) {
  return unloadTextureNative(texture);
}

export function isTextureReady(texture) {
  return isTextureReadyNative(texture);
}

export function getTextureWidth(texture) {
  return getTextureWidthNative(texture);
}

export function getTextureHeight(texture) {
  return getTextureHeightNative(texture);
}

export function drawTexture(texture, x, y, color) {
  return drawTextureNative(texture, x, y, color);
}

export function isKeyPressed(key) {
  return isKeyPressedNative(key);
}

export function isKeyDown(key) {
  return isKeyDownNative(key);
}

export function isMouseButtonDown(button) {
  return isMouseButtonDownNative(button);
}

export function getMouseX() {
  return getMouseXNative();
}

export function getMouseY() {
  return getMouseYNative();
}

export function getMousePosition() {
  return getMousePositionNative();
}

export function isGamepadAvailable(gamepad) {
  return isGamepadAvailableNative(gamepad);
}

export function getGamepadAxisCount(gamepad) {
  return getGamepadAxisCountNative(gamepad);
}

export function isGamepadButtonDown(gamepad, button) {
  return isGamepadButtonDownNative(gamepad, button);
}

export function getGamepadName(gamepad) {
  return getGamepadNameNative(gamepad);
}

export function setTargetFPS(fps) {
  return setTargetFPSNative(fps);
}

export function getFrameTime() {
  return getFrameTimeNative();
}

export function getTime() {
  return getTimeNative();
}

export function setRandomSeed(seed) {
  return setRandomSeedNative(seed);
}

export function getRandomValue(min, max) {
  return getRandomValueNative(min, max);
}
