// @jayess/html/native/html.bind.js
const f = () => {};
export const tokenizeNative = f;
export const parseDocumentNative = f;
export const parseFragmentNative = f;
export const serializeNative = f;
export const createElementNative = f;
export const createTextNative = f;
export const createCommentNative = f;
export const setAttributeNative = f;
export const removeAttributeNative = f;
export const appendChildNative = f;
export const removeChildNative = f;
export const replaceChildNative = f;
export const cloneNodeNative = f;
export const walkDepthFirstNative = f;
export const findByTagNative = f;
export const matchesSelectorNative = f;
export const querySelectorAllNative = f;

export default {
  sources: ["./html.c"],
  exports: {
    tokenizeNative: { symbol: "jayess_html_tokenize", type: "function" },
    parseDocumentNative: { symbol: "jayess_html_parse_document", type: "function" },
    parseFragmentNative: { symbol: "jayess_html_parse_fragment", type: "function" },
    serializeNative: { symbol: "jayess_html_serialize", type: "function" },
    createElementNative: { symbol: "jayess_html_create_element", type: "function" },
    createTextNative: { symbol: "jayess_html_create_text", type: "function" },
    createCommentNative: { symbol: "jayess_html_create_comment", type: "function" },
    setAttributeNative: { symbol: "jayess_html_set_attribute", type: "function" },
    removeAttributeNative: { symbol: "jayess_html_remove_attribute", type: "function" },
    appendChildNative: { symbol: "jayess_html_append_child", type: "function" },
    removeChildNative: { symbol: "jayess_html_remove_child", type: "function" },
    replaceChildNative: { symbol: "jayess_html_replace_child", type: "function" },
    cloneNodeNative: { symbol: "jayess_html_clone_node", type: "function" },
    walkDepthFirstNative: { symbol: "jayess_html_walk_depth_first", type: "function" },
    findByTagNative: { symbol: "jayess_html_find_by_tag", type: "function" },
    matchesSelectorNative: { symbol: "jayess_html_matches_selector", type: "function" },
    querySelectorAllNative: { symbol: "jayess_html_query_selector_all", type: "function" }
  }
};
