// @jayess/curl/index.js
import { createEasyNative } from "./native/curl.bind.js";
import { configureNative } from "./native/curl.bind.js";
import { performNative } from "./native/curl.bind.js";
import { cleanupNative } from "./native/curl.bind.js";
import { createMultiNative } from "./native/curl.bind.js";
import { addHandleNative } from "./native/curl.bind.js";
import { performMultiNative } from "./native/curl.bind.js";
import { cleanupMultiNative } from "./native/curl.bind.js";
import { performStreamNative } from "./native/curl.bind.js";

export function createEasy() {
  return createEasyNative();
}

export function configure(handle, options) {
  return configureNative(handle, options);
}

export function perform(handle) {
  return performNative(handle);
}

export function cleanup(handle) {
  return cleanupNative(handle);
}

export function createMulti() {
  return createMultiNative();
}

export function addHandle(multi, handle) {
  return addHandleNative(multi, handle);
}

export function performMulti(multi) {
  return performMultiNative(multi);
}

export function cleanupMulti(multi) {
  return cleanupMultiNative(multi);
}

export function performStream(handle, onChunk) {
  return performStreamNative(handle, onChunk);
}

export function request(options) {
  var handle = createEasy();
  configure(handle, options);
  var response = perform(handle);
  cleanup(handle);
  return response;
}

export function requestMulti(optionsList) {
  var multi = createMulti();
  var handles = [];
  var index = 0;
  for (index = 0; index < optionsList.length; index = index + 1) {
    var handle = createEasy();
    configure(handle, optionsList[index]);
    addHandle(multi, handle);
    handles.push(handle);
  }
  var responses = performMulti(multi);
  for (index = 0; index < handles.length; index = index + 1) {
    cleanup(handles[index]);
  }
  cleanupMulti(multi);
  return responses;
}

export function requestStream(options, onChunk) {
  var handle = createEasy();
  configure(handle, options);
  var response = performStream(handle, onChunk);
  cleanup(handle);
  return response;
}

async function waitForWorkerResult(workerThread) {
  while (true) {
    var response = workerThread.receive(0);
    if (response != undefined) {
      workerThread.terminate();
      if (response.ok) {
        return response.value;
      }
      throw response.error;
    }
    await timers.sleep(0);
  }
}

async function requestAsyncImpl(options) {
  var workerThread = worker.create((message) => {
    return request(message);
  });
  if (!workerThread.postMessage(options)) {
    workerThread.terminate();
    throw new Error("curl async request could not be scheduled");
  }
  return await waitForWorkerResult(workerThread);
}

async function requestMultiAsyncImpl(optionsList) {
  var workerThread = worker.create((message) => {
    return requestMulti(message);
  });
  if (!workerThread.postMessage(optionsList)) {
    workerThread.terminate();
    throw new Error("curl async multi request could not be scheduled");
  }
  return await waitForWorkerResult(workerThread);
}

export function requestAsync(options) {
  return requestAsyncImpl(options);
}

export function requestMultiAsync(optionsList) {
  return requestMultiAsyncImpl(optionsList);
}
