// @jayess/gtk/index.js
import { initNative } from "./native/gtk.bind.js";
import { createWindowNative } from "./native/gtk.bind.js";
import { createLabelNative } from "./native/gtk.bind.js";
import { createButtonNative } from "./native/gtk.bind.js";
import { createTextInputNative } from "./native/gtk.bind.js";
import { createImageNative } from "./native/gtk.bind.js";
import { createDrawingAreaNative } from "./native/gtk.bind.js";
import { createBoxNative } from "./native/gtk.bind.js";
import { setTitleNative } from "./native/gtk.bind.js";
import { setTextNative } from "./native/gtk.bind.js";
import { addChildNative } from "./native/gtk.bind.js";
import { connectSignalNative } from "./native/gtk.bind.js";
import { emitSignalNative } from "./native/gtk.bind.js";
import { queueDrawNative } from "./native/gtk.bind.js";
import { showNative } from "./native/gtk.bind.js";
import { hideNative } from "./native/gtk.bind.js";
import { pollEventsNative } from "./native/gtk.bind.js";
import { runMainLoopNative } from "./native/gtk.bind.js";
import { quitMainLoopNative } from "./native/gtk.bind.js";
import { destroyWindowNative } from "./native/gtk.bind.js";

export function init() {
  return initNative();
}

export function createWindow() {
  return createWindowNative();
}

export function createLabel(text) {
  return createLabelNative(text);
}

export function createButton(text) {
  return createButtonNative(text);
}

export function createTextInput() {
  return createTextInputNative();
}

export function createImage(path) {
  return createImageNative(path);
}

export function createDrawingArea() {
  return createDrawingAreaNative();
}

export function createBox(vertical, spacing) {
  return createBoxNative(vertical, spacing);
}

export function setTitle(window, title) {
  return setTitleNative(window, title);
}

export function setText(widget, text) {
  return setTextNative(widget, text);
}

export function addChild(parent, child) {
  return addChildNative(parent, child);
}

export function connectSignal(widget, signal, callback) {
  return connectSignalNative(widget, signal, callback);
}

export function emitSignal(widget, signal) {
  return emitSignalNative(widget, signal);
}

export function queueDraw(widget) {
  return queueDrawNative(widget);
}

export function show(window) {
  return showNative(window);
}

export function hide(window) {
  return hideNative(window);
}

export function pollEvents() {
  return pollEventsNative();
}

export function runMainLoop() {
  return runMainLoopNative();
}

export function quitMainLoop() {
  return quitMainLoopNative();
}

export function destroyWindow(window) {
  return destroyWindowNative(window);
}
