// @jayess/html/index.js
import { tokenizeNative, parseDocumentNative, parseFragmentNative, serializeNative, createElementNative, createTextNative, createCommentNative, setAttributeNative, removeAttributeNative, appendChildNative, removeChildNative, replaceChildNative, cloneNodeNative, walkDepthFirstNative, findByTagNative, matchesSelectorNative, querySelectorAllNative } from "./native/html.bind.js";

export function tokenizeHtml(input) {
  return tokenizeNative(input);
}

export function parseHtml(input) {
  return parseDocumentNative(input);
}

export function parseHtmlFragment(input) {
  return parseFragmentNative(input);
}

export function serializeHtml(node) {
  return serializeNative(node, undefined);
}

export function serializeHtmlWithOptions(node, options) {
  return serializeNative(node, options);
}

export function createElement(tagName, attributes, children) {
  return createElementNative(tagName, attributes, children);
}

export function createText(value) {
  return createTextNative(value);
}

export function createComment(value) {
  return createCommentNative(value);
}

export function setAttribute(node, name, value) {
  return setAttributeNative(node, name, value);
}

export function removeAttribute(node, name) {
  return removeAttributeNative(node, name);
}

export function appendChild(node, child) {
  return appendChildNative(node, child);
}

export function removeChild(node, index) {
  return removeChildNative(node, index);
}

export function replaceChild(node, index, child) {
  return replaceChildNative(node, index, child);
}

export function cloneNode(node) {
  return cloneNodeNative(node);
}

export function walkDepthFirst(node) {
  return walkDepthFirstNative(node);
}

export function findByTag(node, tagName) {
  return findByTagNative(node, tagName);
}

export function matchesSelector(node, selector) {
  return matchesSelectorNative(node, selector);
}

export function querySelectorAll(node, selector) {
  return querySelectorAllNative(node, selector);
}
