// @jayess/raylib/native/raylib.bind.js
const f = () => {};
export const setTraceLogLevelNative = f;
export const setConfigFlagsNative = f;
export const initWindowNative = f;
export const closeWindowNative = f;
export const isWindowReadyNative = f;
export const windowShouldCloseNative = f;
export const setWindowTitleNative = f;
export const setWindowSizeNative = f;
export const getScreenWidthNative = f;
export const getScreenHeightNative = f;
export const beginDrawingNative = f;
export const endDrawingNative = f;
export const clearBackgroundNative = f;
export const drawCircleNative = f;
export const drawTextNative = f;
export const genImageColorNative = f;
export const loadImageNative = f;
export const loadImageFromBytesNative = f;
export const unloadImageNative = f;
export const isImageReadyNative = f;
export const getImageWidthNative = f;
export const getImageHeightNative = f;
export const loadTextureNative = f;
export const loadTextureFromImageNative = f;
export const unloadTextureNative = f;
export const isTextureReadyNative = f;
export const getTextureWidthNative = f;
export const getTextureHeightNative = f;
export const drawTextureNative = f;
export const isKeyPressedNative = f;
export const isKeyDownNative = f;
export const isMouseButtonDownNative = f;
export const getMouseXNative = f;
export const getMouseYNative = f;
export const getMousePositionNative = f;
export const isGamepadAvailableNative = f;
export const getGamepadAxisCountNative = f;
export const isGamepadButtonDownNative = f;
export const getGamepadNameNative = f;
export const setTargetFPSNative = f;
export const getFrameTimeNative = f;
export const getTimeNative = f;
export const setRandomSeedNative = f;
export const getRandomValueNative = f;

export default {
  sources: [
    "./raylib.c",
    "../../../../refs/raylib/src/rcore.c",
    "../../../../refs/raylib/src/rshapes.c",
    "../../../../refs/raylib/src/rtextures.c",
    "../../../../refs/raylib/src/rtext.c"
  ],
  includeDirs: ["../../../../refs/raylib/src"],
  cflags: [
    "-DPLATFORM_MEMORY",
    "-DGRAPHICS_API_OPENGL_SOFTWARE",
    "-DEXTERNAL_CONFIG_FLAGS",
    "-DSUPPORT_MODULE_RSHAPES=1",
    "-DSUPPORT_MODULE_RTEXTURES=1",
    "-DSUPPORT_MODULE_RTEXT=1",
    "-DSUPPORT_MODULE_RMODELS=0",
    "-DSUPPORT_MODULE_RAUDIO=0"
  ],
  ldflags: ["-lm"],
  exports: {
    setTraceLogLevelNative: { symbol: "jayess_raylib_set_trace_log_level", type: "function" },
    setConfigFlagsNative: { symbol: "jayess_raylib_set_config_flags", type: "function" },
    initWindowNative: { symbol: "jayess_raylib_init_window", type: "function" },
    closeWindowNative: { symbol: "jayess_raylib_close_window", type: "function" },
    isWindowReadyNative: { symbol: "jayess_raylib_is_window_ready", type: "function" },
    windowShouldCloseNative: { symbol: "jayess_raylib_window_should_close", type: "function" },
    setWindowTitleNative: { symbol: "jayess_raylib_set_window_title", type: "function" },
    setWindowSizeNative: { symbol: "jayess_raylib_set_window_size", type: "function" },
    getScreenWidthNative: { symbol: "jayess_raylib_get_screen_width", type: "function" },
    getScreenHeightNative: { symbol: "jayess_raylib_get_screen_height", type: "function" },
    beginDrawingNative: { symbol: "jayess_raylib_begin_drawing", type: "function" },
    endDrawingNative: { symbol: "jayess_raylib_end_drawing", type: "function" },
    clearBackgroundNative: { symbol: "jayess_raylib_clear_background", type: "function" },
    drawCircleNative: { symbol: "jayess_raylib_draw_circle", type: "function" },
    drawTextNative: { symbol: "jayess_raylib_draw_text", type: "function" },
    genImageColorNative: { symbol: "jayess_raylib_gen_image_color", type: "function" },
    loadImageNative: { symbol: "jayess_raylib_load_image", type: "function" },
    loadImageFromBytesNative: { symbol: "jayess_raylib_load_image_from_bytes", type: "function" },
    unloadImageNative: { symbol: "jayess_raylib_unload_image", type: "function" },
    isImageReadyNative: { symbol: "jayess_raylib_is_image_ready", type: "function" },
    getImageWidthNative: { symbol: "jayess_raylib_get_image_width", type: "function" },
    getImageHeightNative: { symbol: "jayess_raylib_get_image_height", type: "function" },
    loadTextureNative: { symbol: "jayess_raylib_load_texture", type: "function" },
    loadTextureFromImageNative: { symbol: "jayess_raylib_load_texture_from_image", type: "function" },
    unloadTextureNative: { symbol: "jayess_raylib_unload_texture", type: "function" },
    isTextureReadyNative: { symbol: "jayess_raylib_is_texture_ready", type: "function" },
    getTextureWidthNative: { symbol: "jayess_raylib_get_texture_width", type: "function" },
    getTextureHeightNative: { symbol: "jayess_raylib_get_texture_height", type: "function" },
    drawTextureNative: { symbol: "jayess_raylib_draw_texture", type: "function" },
    isKeyPressedNative: { symbol: "jayess_raylib_is_key_pressed", type: "function" },
    isKeyDownNative: { symbol: "jayess_raylib_is_key_down", type: "function" },
    isMouseButtonDownNative: { symbol: "jayess_raylib_is_mouse_button_down", type: "function" },
    getMouseXNative: { symbol: "jayess_raylib_get_mouse_x", type: "function" },
    getMouseYNative: { symbol: "jayess_raylib_get_mouse_y", type: "function" },
    getMousePositionNative: { symbol: "jayess_raylib_get_mouse_position", type: "function" },
    isGamepadAvailableNative: { symbol: "jayess_raylib_is_gamepad_available", type: "function" },
    getGamepadAxisCountNative: { symbol: "jayess_raylib_get_gamepad_axis_count", type: "function" },
    isGamepadButtonDownNative: { symbol: "jayess_raylib_is_gamepad_button_down", type: "function" },
    getGamepadNameNative: { symbol: "jayess_raylib_get_gamepad_name", type: "function" },
    setTargetFPSNative: { symbol: "jayess_raylib_set_target_fps", type: "function" },
    getFrameTimeNative: { symbol: "jayess_raylib_get_frame_time", type: "function" },
    getTimeNative: { symbol: "jayess_raylib_get_time", type: "function" },
    setRandomSeedNative: { symbol: "jayess_raylib_set_random_seed", type: "function" },
    getRandomValueNative: { symbol: "jayess_raylib_get_random_value", type: "function" }
  }
};
