// @jayess/sqlite/index.js
import { openNative } from "./native/sqlite.bind.js";
import { closeNative } from "./native/sqlite.bind.js";
import { execNative } from "./native/sqlite.bind.js";
import { prepareNative } from "./native/sqlite.bind.js";
import { finalizeNative } from "./native/sqlite.bind.js";
import { resetNative } from "./native/sqlite.bind.js";
import { clearBindingsNative } from "./native/sqlite.bind.js";
import { bindNullNative } from "./native/sqlite.bind.js";
import { bindIntegerNative } from "./native/sqlite.bind.js";
import { bindFloatNative } from "./native/sqlite.bind.js";
import { bindStringNative } from "./native/sqlite.bind.js";
import { bindBlobNative } from "./native/sqlite.bind.js";
import { stepObjectNative } from "./native/sqlite.bind.js";
import { stepArrayNative } from "./native/sqlite.bind.js";
import { changesNative } from "./native/sqlite.bind.js";
import { lastInsertRowIdNative } from "./native/sqlite.bind.js";
import { busyTimeoutNative } from "./native/sqlite.bind.js";

export function open(path) {
  return openNative(path);
}

export function close(db) {
  return closeNative(db);
}

export function exec(db, sql) {
  return execNative(db, sql);
}

export function prepare(db, sql) {
  return prepareNative(db, sql);
}

export function finalize(stmt) {
  return finalizeNative(stmt);
}

export function reset(stmt) {
  return resetNative(stmt);
}

export function clearBindings(stmt) {
  return clearBindingsNative(stmt);
}

export function bindNull(stmt, index) {
  return bindNullNative(stmt, index);
}

export function bindInteger(stmt, index, value) {
  return bindIntegerNative(stmt, index, value);
}

export function bindFloat(stmt, index, value) {
  return bindFloatNative(stmt, index, value);
}

export function bindString(stmt, index, value) {
  return bindStringNative(stmt, index, value);
}

export function bindBlob(stmt, index, value) {
  return bindBlobNative(stmt, index, value);
}

export function step(stmt) {
  return stepObjectNative(stmt);
}

export function stepArray(stmt) {
  return stepArrayNative(stmt);
}

export function get(stmt) {
  return step(stmt);
}

export function getArray(stmt) {
  return stepArray(stmt);
}

export function all(stmt) {
  var rows = [];
  var row = step(stmt);
  while (row !== null) {
    rows.push(row);
    row = step(stmt);
  }
  return rows;
}

export function allArray(stmt) {
  var rows = [];
  var row = stepArray(stmt);
  while (row !== null) {
    rows.push(row);
    row = stepArray(stmt);
  }
  return rows;
}

export function run(stmt) {
  while (step(stmt) !== null) {
  }
  return stmt;
}

export function changes(db) {
  return changesNative(db);
}

export function lastInsertRowId(db) {
  return lastInsertRowIdNative(db);
}

export function busyTimeout(db, milliseconds) {
  return busyTimeoutNative(db, milliseconds);
}

export function begin(db) {
  return exec(db, "BEGIN");
}

export function commit(db) {
  return exec(db, "COMMIT");
}

export function rollback(db) {
  return exec(db, "ROLLBACK");
}

export function query(db, sql) {
  var stmt = prepare(db, sql);
  var rows = all(stmt);
  finalize(stmt);
  return rows;
}

export function queryArray(db, sql) {
  var stmt = prepare(db, sql);
  var rows = allArray(stmt);
  finalize(stmt);
  return rows;
}

export function pragma(db, sql) {
  return query(db, "PRAGMA " + sql);
}
