// @jayess/glfw/index.js
import { initNative as glfwInitNative, terminateNative as glfwTerminateNative, createWindowNative as glfwCreateWindowNative, destroyWindowNative as glfwDestroyWindowNative, pollEventsNative as glfwPollEventsNative, swapBuffersNative as glfwSwapBuffersNative, windowShouldCloseNative as glfwWindowShouldCloseNative } from "./native/glfw.bind.js";
import { getTimeNative as glfwGetTimeNative, setTimeNative as glfwSetTimeNative, getWindowSizeNative as glfwGetWindowSizeNative, setWindowSizeNative as glfwSetWindowSizeNative, getFramebufferSizeNative as glfwGetFramebufferSizeNative, setKeyCallbackNative as glfwSetKeyCallbackNative, setMouseButtonCallbackNative as glfwSetMouseButtonCallbackNative, setCursorPosCallbackNative as glfwSetCursorPosCallbackNative, setScrollCallbackNative as glfwSetScrollCallbackNative, simulateKeyEventNative as glfwSimulateKeyEventNative, simulateMouseButtonEventNative as glfwSimulateMouseButtonEventNative, simulateCursorPosEventNative as glfwSimulateCursorPosEventNative, simulateScrollEventNative as glfwSimulateScrollEventNative, setWindowFullscreenNative as glfwSetWindowFullscreenNative, setWindowWindowedNative as glfwSetWindowWindowedNative } from "./native/glfw.bind.js";
import { isJoystickPresentNative as glfwIsJoystickPresentNative, isJoystickGamepadNative as glfwIsJoystickGamepadNative, getJoystickNameNative as glfwGetJoystickNameNative, getGamepadNameNative as glfwGetGamepadNameNative, getGamepadButtonNative as glfwGetGamepadButtonNative } from "./native/glfw.bind.js";

export function init() {
  return glfwInitNative();
}

export function terminate() {
  return glfwTerminateNative();
}

export function createWindow(width, height, title) {
  return glfwCreateWindowNative(width, height, title);
}

export function destroyWindow(window) {
  return glfwDestroyWindowNative(window);
}

export function pollEvents() {
  return glfwPollEventsNative();
}

export function swapBuffers(window) {
  return glfwSwapBuffersNative(window);
}

export function windowShouldClose(window) {
  return glfwWindowShouldCloseNative(window);
}

export function getTime() {
  return glfwGetTimeNative();
}

export function setTime(value) {
  return glfwSetTimeNative(value);
}

export function getWindowSize(window) {
  return glfwGetWindowSizeNative(window);
}

export function setWindowSize(window, width, height) {
  return glfwSetWindowSizeNative(window, width, height);
}

export function getFramebufferSize(window) {
  return glfwGetFramebufferSizeNative(window);
}

export function setKeyCallback(window, callback) {
  return glfwSetKeyCallbackNative(window, callback);
}

export function setMouseButtonCallback(window, callback) {
  return glfwSetMouseButtonCallbackNative(window, callback);
}

export function setCursorPosCallback(window, callback) {
  return glfwSetCursorPosCallbackNative(window, callback);
}

export function setScrollCallback(window, callback) {
  return glfwSetScrollCallbackNative(window, callback);
}

export function simulateKeyEvent(window, key, action, mods) {
  return glfwSimulateKeyEventNative(window, key, action, mods);
}

export function simulateMouseButtonEvent(window, button, action, mods) {
  return glfwSimulateMouseButtonEventNative(window, button, action, mods);
}

export function simulateCursorPosEvent(window, x, y) {
  return glfwSimulateCursorPosEventNative(window, x, y);
}

export function simulateScrollEvent(window, xoffset, yoffset) {
  return glfwSimulateScrollEventNative(window, xoffset, yoffset);
}

export function setWindowFullscreen(window) {
  return glfwSetWindowFullscreenNative(window);
}

export function setWindowWindowed(window, width, height) {
  return glfwSetWindowWindowedNative(window, width, height);
}

export function isJoystickPresent(joystick) {
  return glfwIsJoystickPresentNative(joystick);
}

export function isJoystickGamepad(joystick) {
  return glfwIsJoystickGamepadNative(joystick);
}

export function getJoystickName(joystick) {
  return glfwGetJoystickNameNative(joystick);
}

export function getGamepadName(joystick) {
  return glfwGetGamepadNameNative(joystick);
}

export function getGamepadButton(joystick, button) {
  return glfwGetGamepadButtonNative(joystick, button);
}
