// @jayess/mongoose/index.js
import { createManagerNative } from "./native/mongoose.bind.js";
import { listenNative } from "./native/mongoose.bind.js";
import { listenTlsNative } from "./native/mongoose.bind.js";
import { pollNative } from "./native/mongoose.bind.js";
import { nextEventNative } from "./native/mongoose.bind.js";
import { replyNative } from "./native/mongoose.bind.js";
import { upgradeWebSocketNative } from "./native/mongoose.bind.js";
import { sendWebSocketNative } from "./native/mongoose.bind.js";
import { closeConnectionNative } from "./native/mongoose.bind.js";
import { freeManagerNative } from "./native/mongoose.bind.js";
import { serveStaticNative } from "./native/mongoose.bind.js";
import { startChunkedNative } from "./native/mongoose.bind.js";
import { writeChunkNative } from "./native/mongoose.bind.js";
import { endChunkedNative } from "./native/mongoose.bind.js";

export function createManager() {
  return createManagerNative();
}

export function listenServer(manager, url) {
  return listenNative(manager, url);
}

export function listenTlsServer(manager, url, certPath, keyPath) {
  return listenTlsNative(manager, url, certPath, keyPath);
}

export function pollManager(manager, timeoutMs) {
  return pollNative(manager, timeoutMs);
}

export function nextEvent(manager) {
  return nextEventNative(manager);
}

export function reply(connection, status, headers, body) {
  return replyNative(connection, status, headers, body);
}

export function upgradeWebSocket(event) {
  return upgradeWebSocketNative(event.connection);
}

export function sendWebSocket(connection, data) {
  return sendWebSocketNative(connection, data);
}

export function closeConnection(connection) {
  return closeConnectionNative(connection);
}

export function freeManager(manager) {
  return freeManagerNative(manager);
}

export function createRouter() {
  return [];
}

export function addRoute(router, method, path, handler) {
  router.push({
    method: method,
    path: path,
    handler: handler
  });
  return router;
}

export function get(router, path, handler) {
  return addRoute(router, "GET", path, handler);
}

export function post(router, path, handler) {
  return addRoute(router, "POST", path, handler);
}

export function put(router, path, handler) {
  return addRoute(router, "PUT", path, handler);
}

export function del(router, path, handler) {
  return addRoute(router, "DELETE", path, handler);
}

export function all(router, path, handler) {
  return addRoute(router, "*", path, handler);
}

function routeMatches(route, event) {
  if (route.method !== "*" && route.method !== event.method) {
    return false;
  }
  if (route.path === "*") {
    return true;
  }
  if (route.prefix === true) {
    return event.path === route.path || event.path.slice(0, route.path.length + 1) === route.path + "/";
  }
  return route.path === event.path;
}

export function serveStatic(event, urlPrefix, rootDir) {
  return serveStaticNative(event.connection, event.path, urlPrefix, rootDir);
}

function normalizeEmbeddedPath(path) {
  if (path === undefined || path === null || path === "") {
    return "/";
  }
  if (path.slice(0, 1) === "/") {
    return path;
  }
  return "/" + path;
}

function embeddedContentType(path) {
  if (path.length >= 5 && path.slice(path.length - 5) === ".html") {
    return "text/html; charset=utf-8";
  }
  if (path.length >= 3 && path.slice(path.length - 3) === ".js") {
    return "application/javascript; charset=utf-8";
  }
  if (path.length >= 4 && path.slice(path.length - 4) === ".css") {
    return "text/css; charset=utf-8";
  }
  if (path.length >= 5 && path.slice(path.length - 5) === ".json") {
    return "application/json; charset=utf-8";
  }
  if (path.length >= 4 && path.slice(path.length - 4) === ".svg") {
    return "image/svg+xml";
  }
  if (path.length >= 4 && path.slice(path.length - 4) === ".txt") {
    return "text/plain; charset=utf-8";
  }
  return "application/octet-stream";
}

function cloneHeaders(headers) {
  var result = {};
  var keys = Object.keys(headers);
  var i = 0;
  while (i < keys.length) {
    var key = keys[i];
    result[key] = headers[key];
    i = i + 1;
  }
  return result;
}

function findEmbeddedAsset(assets, targetPath) {
  var i = 0;
  while (i < assets.length) {
    if (normalizeEmbeddedPath(assets[i].path) === targetPath) {
      return assets[i];
    }
    i = i + 1;
  }
  return undefined;
}

export function createEmbeddedApp(files, fallbackPath) {
  return {
    assets: files,
    files: files,
    fallbackPath: fallbackPath !== undefined ? normalizeEmbeddedPath(fallbackPath) : undefined
  };
}

export function serveEmbeddedApp(event, urlPrefix, app, fallbackPathOverride) {
  var prefix = urlPrefix !== undefined && urlPrefix !== null && urlPrefix !== "" ? urlPrefix : "/";
  var assets = app.files !== undefined ? app.files : (app.assets !== undefined ? app.assets : app);
  var fallbackPath = fallbackPathOverride !== undefined ? normalizeEmbeddedPath(fallbackPathOverride) : app.fallbackPath;
  var method = event.method;
  var relativePath = "";
  var asset = undefined;
  var assetPath = "";
  var body = "";
  var headers = {};
  var status = 200;
  if (method === "get") { method = "GET"; }
  if (method === "head") { method = "HEAD"; }
  if (method !== "GET" && method !== "HEAD") {
    return false;
  }
  if (prefix === "/") {
    relativePath = event.path;
  } else if (event.path === prefix || event.path === prefix + "/") {
    relativePath = "/";
  } else if (event.path.slice(0, prefix.length + 1) === prefix + "/") {
    relativePath = event.path.slice(prefix.length);
  } else {
    return false;
  }
  relativePath = normalizeEmbeddedPath(relativePath);
  if (relativePath === "/") {
    relativePath = "/index.html";
  }
  assetPath = relativePath;
  asset = findEmbeddedAsset(assets, assetPath);
  if (asset === undefined && fallbackPath !== undefined) {
    assetPath = fallbackPath;
    asset = findEmbeddedAsset(assets, assetPath);
  }
  if (asset === undefined) {
    return false;
  }
  status = asset.status !== undefined ? asset.status : 200;
  body = asset.body !== undefined ? asset.body : asset;
  headers = asset.headers !== undefined ? cloneHeaders(asset.headers) : {};
  if (headers["Content-Type"] === undefined) {
    headers["Content-Type"] = asset.contentType !== undefined ? asset.contentType : embeddedContentType(assetPath);
  }
  reply(event.connection, status, headers, method === "HEAD" ? "" : body);
  return true;
}

export function startChunked(event, status, headers) {
  return startChunkedNative(event.connection, status, headers);
}

export function writeChunk(stream, chunk) {
  return writeChunkNative(stream, chunk);
}

export function endChunked(stream, finalChunk) {
  return endChunkedNative(stream, finalChunk);
}

export function dispatch(router, event) {
  var i = 0;
  while (i < router.length) {
    var route = router[i];
    if (routeMatches(route, event)) {
      var response = route.handler(event);
      if (response === undefined || response === null) {
        return false;
      }
      var status = response.status !== undefined ? response.status : 200;
      var headers = response.headers !== undefined ? response.headers : {};
      var body = response.body !== undefined ? response.body : "";
      reply(event.connection, status, headers, body);
      return true;
    }
    i = i + 1;
  }
  return false;
}
