// @jayess/glfw/native/glfw.bind.js
const f = () => {};
export const initNative = f;
export const terminateNative = f;
export const createWindowNative = f;
export const destroyWindowNative = f;
export const pollEventsNative = f;
export const swapBuffersNative = f;
export const windowShouldCloseNative = f;
export const getTimeNative = f;
export const setTimeNative = f;
export const getWindowSizeNative = f;
export const setWindowSizeNative = f;
export const getFramebufferSizeNative = f;
export const setKeyCallbackNative = f;
export const setMouseButtonCallbackNative = f;
export const setCursorPosCallbackNative = f;
export const setScrollCallbackNative = f;
export const simulateKeyEventNative = f;
export const simulateMouseButtonEventNative = f;
export const simulateCursorPosEventNative = f;
export const simulateScrollEventNative = f;
export const setWindowFullscreenNative = f;
export const setWindowWindowedNative = f;
export const isJoystickPresentNative = f;
export const isJoystickGamepadNative = f;
export const getJoystickNameNative = f;
export const getGamepadNameNative = f;
export const getGamepadButtonNative = f;

export default {
  sources: [
    "./glfw.c",
    "../../../../refs/glfw/src/context.c",
    "../../../../refs/glfw/src/init.c",
    "../../../../refs/glfw/src/input.c",
    "../../../../refs/glfw/src/monitor.c",
    "../../../../refs/glfw/src/platform.c",
    "../../../../refs/glfw/src/vulkan.c",
    "../../../../refs/glfw/src/window.c",
    "../../../../refs/glfw/src/egl_context.c",
    "../../../../refs/glfw/src/osmesa_context.c",
    "../../../../refs/glfw/src/null_init.c",
    "../../../../refs/glfw/src/null_monitor.c",
    "../../../../refs/glfw/src/null_window.c",
    "../../../../refs/glfw/src/null_joystick.c",
    "../../../../refs/glfw/src/posix_module.c",
    "../../../../refs/glfw/src/posix_thread.c",
    "../../../../refs/glfw/src/posix_time.c"
  ],
  includeDirs: ["../../../../refs/glfw/include", "../../../../refs/glfw/src"],
  cflags: [],
  ldflags: ["-pthread", "-ldl"],
  exports: {
    initNative: { symbol: "jayess_glfw_init", type: "function" },
    terminateNative: { symbol: "jayess_glfw_terminate", type: "function" },
    createWindowNative: { symbol: "jayess_glfw_create_window", type: "function" },
    destroyWindowNative: { symbol: "jayess_glfw_destroy_window", type: "function" },
    pollEventsNative: { symbol: "jayess_glfw_poll_events", type: "function" },
    swapBuffersNative: { symbol: "jayess_glfw_swap_buffers", type: "function" },
    windowShouldCloseNative: { symbol: "jayess_glfw_window_should_close", type: "function" },
    getTimeNative: { symbol: "jayess_glfw_get_time", type: "function" },
    setTimeNative: { symbol: "jayess_glfw_set_time", type: "function" },
    getWindowSizeNative: { symbol: "jayess_glfw_get_window_size", type: "function" },
    setWindowSizeNative: { symbol: "jayess_glfw_set_window_size", type: "function" },
    getFramebufferSizeNative: { symbol: "jayess_glfw_get_framebuffer_size", type: "function" },
    setKeyCallbackNative: { symbol: "jayess_glfw_set_key_callback", type: "function" },
    setMouseButtonCallbackNative: { symbol: "jayess_glfw_set_mouse_button_callback", type: "function" },
    setCursorPosCallbackNative: { symbol: "jayess_glfw_set_cursor_pos_callback", type: "function" },
    setScrollCallbackNative: { symbol: "jayess_glfw_set_scroll_callback", type: "function" },
    simulateKeyEventNative: { symbol: "jayess_glfw_simulate_key_event", type: "function" },
    simulateMouseButtonEventNative: { symbol: "jayess_glfw_simulate_mouse_button_event", type: "function" },
    simulateCursorPosEventNative: { symbol: "jayess_glfw_simulate_cursor_pos_event", type: "function" },
    simulateScrollEventNative: { symbol: "jayess_glfw_simulate_scroll_event", type: "function" },
    setWindowFullscreenNative: { symbol: "jayess_glfw_set_window_fullscreen", type: "function" },
    setWindowWindowedNative: { symbol: "jayess_glfw_set_window_windowed", type: "function" },
    isJoystickPresentNative: { symbol: "jayess_glfw_is_joystick_present", type: "function" },
    isJoystickGamepadNative: { symbol: "jayess_glfw_is_joystick_gamepad", type: "function" },
    getJoystickNameNative: { symbol: "jayess_glfw_get_joystick_name", type: "function" },
    getGamepadNameNative: { symbol: "jayess_glfw_get_gamepad_name", type: "function" },
    getGamepadButtonNative: { symbol: "jayess_glfw_get_gamepad_button", type: "function" }
  }
};
