// @jayess/openssl/native/openssl.bind.js
const f = () => {};
export const randomBytesNative = f;
export const versionNative = f;
export const supportsHashNative = f;
export const supportsCipherNative = f;
export const hashNative = f;
export const hmacNative = f;
export const encryptNative = f;
export const decryptNative = f;
export const generateKeyPairNative = f;
export const publicEncryptNative = f;
export const privateDecryptNative = f;
export const signNative = f;
export const verifyNative = f;

export default {
  sources: ["./openssl.c"],
  includeDirs: ["."],
  cflags: [],
  ldflags: ["-lssl", "-lcrypto"],
  exports: {
    randomBytesNative: { symbol: "jayess_openssl_random_bytes_native", type: "function" },
    versionNative: { symbol: "jayess_openssl_version_native", type: "function" },
    supportsHashNative: { symbol: "jayess_openssl_supports_hash_native", type: "function" },
    supportsCipherNative: { symbol: "jayess_openssl_supports_cipher_native", type: "function" },
    hashNative: { symbol: "jayess_openssl_hash_native", type: "function" },
    hmacNative: { symbol: "jayess_openssl_hmac_native", type: "function" },
    encryptNative: { symbol: "jayess_openssl_encrypt_native", type: "function" },
    decryptNative: { symbol: "jayess_openssl_decrypt_native", type: "function" },
    generateKeyPairNative: { symbol: "jayess_openssl_generate_key_pair_native", type: "function" },
    publicEncryptNative: { symbol: "jayess_openssl_public_encrypt_native", type: "function" },
    privateDecryptNative: { symbol: "jayess_openssl_private_decrypt_native", type: "function" },
    signNative: { symbol: "jayess_openssl_sign_native", type: "function" },
    verifyNative: { symbol: "jayess_openssl_verify_native", type: "function" }
  }
};
