// @jayess/sqlite/native/sqlite.bind.js
const f = () => {};
export const openNative = f;
export const closeNative = f;
export const execNative = f;
export const prepareNative = f;
export const finalizeNative = f;
export const resetNative = f;
export const clearBindingsNative = f;
export const bindNullNative = f;
export const bindIntegerNative = f;
export const bindFloatNative = f;
export const bindStringNative = f;
export const bindBlobNative = f;
export const stepObjectNative = f;
export const stepArrayNative = f;
export const changesNative = f;
export const lastInsertRowIdNative = f;
export const busyTimeoutNative = f;

export default {
  sources: ["./sqlite.c", "../../../../refs/sqlite/sqlite3.c"],
  includeDirs: [".", "../../../../refs/sqlite"],
  cflags: [],
  ldflags: [],
  exports: {
    openNative: { symbol: "jayess_sqlite_open_native", type: "function" },
    closeNative: { symbol: "jayess_sqlite_close_native", type: "function" },
    execNative: { symbol: "jayess_sqlite_exec_native", type: "function" },
    prepareNative: { symbol: "jayess_sqlite_prepare_native", type: "function" },
    finalizeNative: { symbol: "jayess_sqlite_finalize_native", type: "function" },
    resetNative: { symbol: "jayess_sqlite_reset_native", type: "function" },
    clearBindingsNative: { symbol: "jayess_sqlite_clear_bindings_native", type: "function" },
    bindNullNative: { symbol: "jayess_sqlite_bind_null_native", type: "function" },
    bindIntegerNative: { symbol: "jayess_sqlite_bind_integer_native", type: "function" },
    bindFloatNative: { symbol: "jayess_sqlite_bind_float_native", type: "function" },
    bindStringNative: { symbol: "jayess_sqlite_bind_string_native", type: "function" },
    bindBlobNative: { symbol: "jayess_sqlite_bind_blob_native", type: "function" },
    stepObjectNative: { symbol: "jayess_sqlite_step_object_native", type: "function" },
    stepArrayNative: { symbol: "jayess_sqlite_step_array_native", type: "function" },
    changesNative: { symbol: "jayess_sqlite_changes_native", type: "function" },
    lastInsertRowIdNative: { symbol: "jayess_sqlite_last_insert_rowid_native", type: "function" },
    busyTimeoutNative: { symbol: "jayess_sqlite_busy_timeout_native", type: "function" }
  }
};
