// @jayess/webview/native/webview.bind.js
const f = () => {};
export const createWindowNative = f;
export const destroyWindowNative = f;
export const setTitleNative = f;
export const setSizeNative = f;
export const showNative = f;
export const hideNative = f;
export const setHtmlNative = f;
export const navigateNative = f;
export const initJsNative = f;
export const evalJsNative = f;
export const bindNative = f;
export const unbindNative = f;
export const nextBindingEventNative = f;
export const returnBindingNative = f;
export const runNative = f;
export const terminateNative = f;

export default {
  sources: ["./webview.cpp"],
  includeDirs: ["../../../../refs/webview/core/include"],
  cflags: ["-std=c++14"],
  ldflags: [],
  platforms: {
    linux: {
      ldflags: [
        "-lstdc++",
        "-ldl",
        "-lgtk-3",
        "-lwebkit2gtk-4.1",
        "-lgobject-2.0",
        "-lglib-2.0",
        "-lgio-2.0"
      ]
    },
    darwin: {
      ldflags: [
        "-lstdc++",
        "-framework",
        "Cocoa",
        "-framework",
        "WebKit"
      ]
    },
    windows: {
      ldflags: [
        "-lstdc++",
        "-lole32",
        "-lcomctl32",
        "-luser32",
        "-lshell32",
        "-ladvapi32"
      ]
    }
  },
  exports: {
    createWindowNative: { symbol: "jayess_webview_create_window", type: "function" },
    destroyWindowNative: { symbol: "jayess_webview_destroy_window", type: "function" },
    setTitleNative: { symbol: "jayess_webview_set_title", type: "function" },
    setSizeNative: { symbol: "jayess_webview_set_size", type: "function" },
    showNative: { symbol: "jayess_webview_show", type: "function" },
    hideNative: { symbol: "jayess_webview_hide", type: "function" },
    setHtmlNative: { symbol: "jayess_webview_set_html", type: "function" },
    navigateNative: { symbol: "jayess_webview_navigate", type: "function" },
    initJsNative: { symbol: "jayess_webview_init_js", type: "function" },
    evalJsNative: { symbol: "jayess_webview_eval_js", type: "function" },
    bindNative: { symbol: "jayess_webview_bind", type: "function" },
    unbindNative: { symbol: "jayess_webview_unbind", type: "function" },
    nextBindingEventNative: { symbol: "jayess_webview_next_binding_event", type: "function" },
    returnBindingNative: { symbol: "jayess_webview_return_binding", type: "function" },
    runNative: { symbol: "jayess_webview_run", type: "function" },
    terminateNative: { symbol: "jayess_webview_terminate", type: "function" }
  }
};
