// @jayess/gtk/native/gtk.bind.js
const f = () => {};
export const initNative = f;
export const createWindowNative = f;
export const createLabelNative = f;
export const createButtonNative = f;
export const createTextInputNative = f;
export const createImageNative = f;
export const createDrawingAreaNative = f;
export const createBoxNative = f;
export const setTitleNative = f;
export const setTextNative = f;
export const addChildNative = f;
export const connectSignalNative = f;
export const emitSignalNative = f;
export const queueDrawNative = f;
export const showNative = f;
export const hideNative = f;
export const pollEventsNative = f;
export const runMainLoopNative = f;
export const quitMainLoopNative = f;
export const destroyWindowNative = f;

export default {
  sources: ["./gtk.c"],
  includeDirs: [],
  cflags: [],
  ldflags: [],
  platforms: {
    linux: {
      ldflags: ["-lgtk-3", "-lgobject-2.0", "-lglib-2.0", "-lgio-2.0"]
    },
    darwin: {
      ldflags: ["-lgtk-3", "-lgobject-2.0", "-lglib-2.0", "-lgio-2.0", "-framework", "Cocoa"]
    },
    windows: {
      ldflags: ["-lgtk-3", "-lgobject-2.0", "-lglib-2.0", "-lgio-2.0", "-lole32", "-lcomctl32", "-luser32"]
    }
  },
  exports: {
    initNative: { symbol: "jayess_gtk_init", type: "function" },
    createWindowNative: { symbol: "jayess_gtk_create_window", type: "function" },
    createLabelNative: { symbol: "jayess_gtk_create_label", type: "function" },
    createButtonNative: { symbol: "jayess_gtk_create_button", type: "function" },
    createTextInputNative: { symbol: "jayess_gtk_create_text_input", type: "function" },
    createImageNative: { symbol: "jayess_gtk_create_image", type: "function" },
    createDrawingAreaNative: { symbol: "jayess_gtk_create_drawing_area", type: "function" },
    createBoxNative: { symbol: "jayess_gtk_create_box", type: "function" },
    setTitleNative: { symbol: "jayess_gtk_set_title", type: "function" },
    setTextNative: { symbol: "jayess_gtk_set_text", type: "function" },
    addChildNative: { symbol: "jayess_gtk_add_child", type: "function" },
    connectSignalNative: { symbol: "jayess_gtk_connect_signal", type: "function" },
    emitSignalNative: { symbol: "jayess_gtk_emit_signal", type: "function" },
    queueDrawNative: { symbol: "jayess_gtk_queue_draw", type: "function" },
    showNative: { symbol: "jayess_gtk_show", type: "function" },
    hideNative: { symbol: "jayess_gtk_hide", type: "function" },
    pollEventsNative: { symbol: "jayess_gtk_poll_events", type: "function" },
    runMainLoopNative: { symbol: "jayess_gtk_run_main_loop", type: "function" },
    quitMainLoopNative: { symbol: "jayess_gtk_quit_main_loop", type: "function" },
    destroyWindowNative: { symbol: "jayess_gtk_destroy_window", type: "function" }
  }
};
