// @jayess/libuv/index.js
import { createLoopNative } from "./native/libuv.bind.js";
import { runNative } from "./native/libuv.bind.js";
import { runOnceNative } from "./native/libuv.bind.js";
import { stopNative } from "./native/libuv.bind.js";
import { closeLoopNative } from "./native/libuv.bind.js";
import { scheduleStopNative } from "./native/libuv.bind.js";
import { scheduleCallbackNative } from "./native/libuv.bind.js";
import { readFileNative } from "./native/libuv.bind.js";
import { watchSignalNative } from "./native/libuv.bind.js";
import { closeSignalWatcherNative } from "./native/libuv.bind.js";
import { watchPathNative } from "./native/libuv.bind.js";
import { closePathWatcherNative } from "./native/libuv.bind.js";
import { spawnProcessNative } from "./native/libuv.bind.js";
import { closeProcessNative } from "./native/libuv.bind.js";
import { createUDPNative } from "./native/libuv.bind.js";
import { bindUDPNative } from "./native/libuv.bind.js";
import { recvUDPNative } from "./native/libuv.bind.js";
import { sendUDPNative } from "./native/libuv.bind.js";
import { closeUDPNative } from "./native/libuv.bind.js";
import { createTCPServerNative } from "./native/libuv.bind.js";
import { listenTCPNative } from "./native/libuv.bind.js";
import { acceptTCPNative } from "./native/libuv.bind.js";
import { closeTCPServerNative } from "./native/libuv.bind.js";
import { createTCPClientNative } from "./native/libuv.bind.js";
import { connectTCPNative } from "./native/libuv.bind.js";
import { readTCPNative } from "./native/libuv.bind.js";
import { writeTCPNative } from "./native/libuv.bind.js";
import { closeTCPClientNative } from "./native/libuv.bind.js";
import { nowNative } from "./native/libuv.bind.js";

export function createLoop() {
  return createLoopNative();
}

export function run(loop) {
  return runNative(loop);
}

export function runOnce(loop) {
  return runOnceNative(loop);
}

export function stop(loop) {
  return stopNative(loop);
}

export function closeLoop(loop) {
  return closeLoopNative(loop);
}

export function scheduleStop(loop, timeoutMs) {
  return scheduleStopNative(loop, timeoutMs);
}

export function scheduleCallback(loop, timeoutMs, callback) {
  return scheduleCallbackNative(loop, timeoutMs, callback);
}

export function readFile(loop, path, callback) {
  return readFileNative(loop, path, callback);
}

export function watchSignal(loop, signal, callback) {
  return watchSignalNative(loop, signal, callback);
}

export function closeSignalWatcher(watcher) {
  return closeSignalWatcherNative(watcher);
}

export function watchPath(loop, path, callback) {
  return watchPathNative(loop, path, callback);
}

export function closePathWatcher(watcher) {
  return closePathWatcherNative(watcher);
}

export function spawnProcess(loop, file, args, callback) {
  return spawnProcessNative(loop, file, args, callback);
}

export function closeProcess(process) {
  return closeProcessNative(process);
}

export function createUDP(loop) {
  return createUDPNative(loop);
}

export function bindUDP(socket, host, port) {
  return bindUDPNative(socket, host, port);
}

export function recvUDP(socket, callback) {
  return recvUDPNative(socket, callback);
}

export function sendUDP(socket, host, port, data) {
  return sendUDPNative(socket, host, port, data);
}

export function closeUDP(socket) {
  return closeUDPNative(socket);
}

export function createTCPServer(loop) {
  return createTCPServerNative(loop);
}

export function listenTCP(server, host, port, callback) {
  return listenTCPNative(server, host, port, callback);
}

export function acceptTCP(server) {
  return acceptTCPNative(server);
}

export function closeTCPServer(server) {
  return closeTCPServerNative(server);
}

export function createTCPClient(loop) {
  return createTCPClientNative(loop);
}

export function connectTCP(client, host, port, callback) {
  return connectTCPNative(client, host, port, callback);
}

export function readTCP(client, callback) {
  return readTCPNative(client, callback);
}

export function writeTCP(client, data) {
  return writeTCPNative(client, data);
}

export function closeTCPClient(client) {
  return closeTCPClientNative(client);
}

export function now(loop) {
  return nowNative(loop);
}
