// @jayess/audio/native/audio.bind.js
const f = () => {};
export const createContextNative = f;
export const destroyContextNative = f;
export const backendIdNative = f;
export const maxChannelCountNative = f;
export const listOutputDevicesNative = f;
export const listInputDevicesNative = f;
export const preferredSampleRateNative = f;
export const minLatencyNative = f;
export const createPlaybackStreamNative = f;
export const startPlaybackStreamNative = f;
export const pausePlaybackStreamNative = f;
export const stopPlaybackStreamNative = f;
export const submitPlaybackSamplesNative = f;
export const playbackStatsNative = f;
export const closePlaybackStreamNative = f;
export const nextStreamEventNative = f;
export const createCaptureStreamNative = f;
export const startCaptureStreamNative = f;
export const stopCaptureStreamNative = f;
export const readCapturedSamplesNative = f;
export const captureStatsNative = f;
export const closeCaptureStreamNative = f;
export const loadWavNative = f;
export const loadOggNative = f;
export const loadMp3Native = f;
export const loadFlacNative = f;

export default {
  sources: ["./audio.c", "../../../../refs/miniaudio/miniaudio.c"],
  includeDirs: [".", "../../../../refs/cubeb/include", "../../../../refs/miniaudio", "../../../../refs/miniaudio/extras"],
  cflags: ["-DMA_NO_DEVICE_IO", "-DMA_NO_THREADING"],
  ldflags: ["-lcubeb", "-pthread", "-ldl", "-lm"],
  exports: {
    createContextNative: { symbol: "jayess_audio_create_context", type: "function" },
    destroyContextNative: { symbol: "jayess_audio_destroy_context", type: "function" },
    backendIdNative: { symbol: "jayess_audio_backend_id", type: "function" },
    maxChannelCountNative: { symbol: "jayess_audio_max_channel_count", type: "function" },
    listOutputDevicesNative: { symbol: "jayess_audio_list_output_devices", type: "function" },
    listInputDevicesNative: { symbol: "jayess_audio_list_input_devices", type: "function" },
    preferredSampleRateNative: { symbol: "jayess_audio_preferred_sample_rate", type: "function" },
    minLatencyNative: { symbol: "jayess_audio_min_latency", type: "function" },
    createPlaybackStreamNative: { symbol: "jayess_audio_create_playback_stream", type: "function" },
    startPlaybackStreamNative: { symbol: "jayess_audio_start_playback_stream", type: "function" },
    pausePlaybackStreamNative: { symbol: "jayess_audio_pause_playback_stream", type: "function" },
    stopPlaybackStreamNative: { symbol: "jayess_audio_stop_playback_stream", type: "function" },
    submitPlaybackSamplesNative: { symbol: "jayess_audio_submit_playback_samples", type: "function" },
    playbackStatsNative: { symbol: "jayess_audio_playback_stats", type: "function" },
    closePlaybackStreamNative: { symbol: "jayess_audio_close_playback_stream", type: "function" },
    nextStreamEventNative: { symbol: "jayess_audio_next_stream_event", type: "function" },
    createCaptureStreamNative: { symbol: "jayess_audio_create_capture_stream", type: "function" },
    startCaptureStreamNative: { symbol: "jayess_audio_start_capture_stream", type: "function" },
    stopCaptureStreamNative: { symbol: "jayess_audio_stop_capture_stream", type: "function" },
    readCapturedSamplesNative: { symbol: "jayess_audio_read_captured_samples", type: "function" },
    captureStatsNative: { symbol: "jayess_audio_capture_stats", type: "function" },
    closeCaptureStreamNative: { symbol: "jayess_audio_close_capture_stream", type: "function" },
    loadWavNative: { symbol: "jayess_audio_load_wav", type: "function" },
    loadOggNative: { symbol: "jayess_audio_load_ogg", type: "function" },
    loadMp3Native: { symbol: "jayess_audio_load_mp3", type: "function" },
    loadFlacNative: { symbol: "jayess_audio_load_flac", type: "function" }
  }
};
