// @jayess/mongoose/native/mongoose.bind.js
const f = () => {};
export const createManagerNative = f;
export const listenNative = f;
export const listenTlsNative = f;
export const pollNative = f;
export const nextEventNative = f;
export const replyNative = f;
export const upgradeWebSocketNative = f;
export const sendWebSocketNative = f;
export const closeConnectionNative = f;
export const freeManagerNative = f;
export const serveStaticNative = f;
export const startChunkedNative = f;
export const writeChunkNative = f;
export const endChunkedNative = f;

export default {
  sources: [
    "./mongoose.c",
    "../../../../refs/mongoose/mongoose.c"
  ],
  includeDirs: [
    ".",
    "../../../../refs/mongoose"
  ],
  cflags: ["-DMG_TLS=MG_TLS_BUILTIN"],
  ldflags: [],
  exports: {
    createManagerNative: { symbol: "jayess_mongoose_create_manager", type: "function" },
    listenNative: { symbol: "jayess_mongoose_listen", type: "function" },
    listenTlsNative: { symbol: "jayess_mongoose_listen_tls", type: "function" },
    pollNative: { symbol: "jayess_mongoose_poll", type: "function" },
    nextEventNative: { symbol: "jayess_mongoose_next_event", type: "function" },
    replyNative: { symbol: "jayess_mongoose_reply", type: "function" },
    upgradeWebSocketNative: { symbol: "jayess_mongoose_upgrade_websocket", type: "function" },
    sendWebSocketNative: { symbol: "jayess_mongoose_send_websocket", type: "function" },
    closeConnectionNative: { symbol: "jayess_mongoose_close_connection", type: "function" },
    freeManagerNative: { symbol: "jayess_mongoose_free_manager", type: "function" },
    serveStaticNative: { symbol: "jayess_mongoose_serve_static", type: "function" },
    startChunkedNative: { symbol: "jayess_mongoose_start_chunked", type: "function" },
    writeChunkNative: { symbol: "jayess_mongoose_write_chunk", type: "function" },
    endChunkedNative: { symbol: "jayess_mongoose_end_chunked", type: "function" }
  }
};
