// @jayess/audio/index.js
import { createContextNative, destroyContextNative, backendIdNative, maxChannelCountNative } from "./native/audio.bind.js";
import { listOutputDevicesNative, listInputDevicesNative } from "./native/audio.bind.js";
import { preferredSampleRateNative, minLatencyNative } from "./native/audio.bind.js";
import { createPlaybackStreamNative, startPlaybackStreamNative, pausePlaybackStreamNative, stopPlaybackStreamNative } from "./native/audio.bind.js";
import { submitPlaybackSamplesNative, playbackStatsNative, closePlaybackStreamNative, nextStreamEventNative } from "./native/audio.bind.js";
import { createCaptureStreamNative, startCaptureStreamNative, stopCaptureStreamNative, readCapturedSamplesNative, captureStatsNative, closeCaptureStreamNative } from "./native/audio.bind.js";
import { loadWavNative } from "./native/audio.bind.js";
import { loadOggNative, loadMp3Native, loadFlacNative } from "./native/audio.bind.js";

export function createContext(name, backendName) {
  return createContextNative(name, backendName);
}

export function destroyContext(context) {
  return destroyContextNative(context);
}

export function backendId(context) {
  return backendIdNative(context);
}

export function maxChannelCount(context) {
  return maxChannelCountNative(context);
}

export function listOutputDevices(context) {
  return listOutputDevicesNative(context);
}

export function listInputDevices(context) {
  return listInputDevicesNative(context);
}

export function preferredSampleRate(context) {
  return preferredSampleRateNative(context);
}

export function minLatency(context, options) {
  return minLatencyNative(context, options);
}

export function createPlaybackStream(context, options) {
  return createPlaybackStreamNative(context, options);
}

export function startPlaybackStream(stream) {
  return startPlaybackStreamNative(stream);
}

export function pausePlaybackStream(stream) {
  return pausePlaybackStreamNative(stream);
}

export function stopPlaybackStream(stream) {
  return stopPlaybackStreamNative(stream);
}

export function submitPlaybackSamples(stream, samples) {
  return submitPlaybackSamplesNative(stream, samples);
}

export function playbackStats(stream) {
  return playbackStatsNative(stream);
}

export function closePlaybackStream(stream) {
  return closePlaybackStreamNative(stream);
}

export function nextStreamEvent(stream) {
  return nextStreamEventNative(stream);
}

export function createCaptureStream(context, options) {
  return createCaptureStreamNative(context, options);
}

export function startCaptureStream(stream) {
  return startCaptureStreamNative(stream);
}

export function stopCaptureStream(stream) {
  return stopCaptureStreamNative(stream);
}

export function readCapturedSamples(stream, frames) {
  return readCapturedSamplesNative(stream, frames);
}

export function captureStats(stream) {
  return captureStatsNative(stream);
}

export function closeCaptureStream(stream) {
  return closeCaptureStreamNative(stream);
}

export function loadWav(path) {
  return loadWavNative(path);
}

export function loadOgg(path) {
  return loadOggNative(path);
}

export function loadMp3(path) {
  return loadMp3Native(path);
}

export function loadFlac(path) {
  return loadFlacNative(path);
}
