// @jayess/webview/index.js
import { createWindowNative as webviewCreateWindowNative } from "./native/webview.bind.js";
import { destroyWindowNative as webviewDestroyWindowNative } from "./native/webview.bind.js";
import { setTitleNative as webviewSetTitleNative } from "./native/webview.bind.js";
import { setSizeNative as webviewSetSizeNative } from "./native/webview.bind.js";
import { showNative as webviewShowNative } from "./native/webview.bind.js";
import { hideNative as webviewHideNative } from "./native/webview.bind.js";
import { setHtmlNative as webviewSetHtmlNative } from "./native/webview.bind.js";
import { navigateNative as webviewNavigateNative } from "./native/webview.bind.js";
import { initJsNative as webviewInitJsNative } from "./native/webview.bind.js";
import { evalJsNative as webviewEvalJsNative } from "./native/webview.bind.js";
import { bindNative as webviewBindNative } from "./native/webview.bind.js";
import { unbindNative as webviewUnbindNative } from "./native/webview.bind.js";
import { nextBindingEventNative as webviewNextBindingEventNative } from "./native/webview.bind.js";
import { returnBindingNative as webviewReturnBindingNative } from "./native/webview.bind.js";
import { runNative as webviewRunNative } from "./native/webview.bind.js";
import { terminateNative as webviewTerminateNative } from "./native/webview.bind.js";

export function createWindow(debug) {
  return webviewCreateWindowNative(debug);
}

export function destroyWindow(view) {
  return webviewDestroyWindowNative(view);
}

export function setTitle(view, title) {
  return webviewSetTitleNative(view, title);
}

export function setSize(view, width, height) {
  return webviewSetSizeNative(view, width, height);
}

export function show(view) {
  return webviewShowNative(view);
}

export function hide(view) {
  return webviewHideNative(view);
}

export function setHtml(view, html) {
  return webviewSetHtmlNative(view, html);
}

export function navigate(view, url) {
  return webviewNavigateNative(view, url);
}

export function loadFile(view, path) {
  return webviewNavigateNative(view, "file://" + path);
}

export function initJs(view, source) {
  return webviewInitJsNative(view, source);
}

export function evalJs(view, source) {
  return webviewEvalJsNative(view, source);
}

export function bind(view, name) {
  return webviewBindNative(view, name);
}

export function unbind(view, name) {
  return webviewUnbindNative(view, name);
}

export function nextBindingEvent(view) {
  return webviewNextBindingEventNative(view);
}

export function returnBinding(view, id, status, result) {
  return webviewReturnBindingNative(view, id, status, result);
}

export function run(view) {
  return webviewRunNative(view);
}

export function terminate(view) {
  return webviewTerminateNative(view);
}
