// @jayess/curl/native/curl.bind.js
const f = () => {};
export const createEasyNative = f;
export const configureNative = f;
export const performNative = f;
export const cleanupNative = f;
export const createMultiNative = f;
export const addHandleNative = f;
export const performMultiNative = f;
export const cleanupMultiNative = f;
export const performStreamNative = f;

export default {
  sources: ["./curl.c"],
  includeDirs: ["../../../../refs/curl/include"],
  cflags: [],
  ldflags: [],
  platforms: {
    linux: {
      ldflags: ["/lib/x86_64-linux-gnu/libcurl.so.4"]
    }
  },
  exports: {
    createEasyNative: { symbol: "jayess_curl_create_easy_native", type: "function" },
    configureNative: { symbol: "jayess_curl_configure_native", type: "function" },
    performNative: { symbol: "jayess_curl_perform_native", type: "function" },
    cleanupNative: { symbol: "jayess_curl_cleanup_native", type: "function" },
    createMultiNative: { symbol: "jayess_curl_create_multi_native", type: "function" },
    addHandleNative: { symbol: "jayess_curl_add_handle_native", type: "function" },
    performMultiNative: { symbol: "jayess_curl_perform_multi_native", type: "function" },
    cleanupMultiNative: { symbol: "jayess_curl_cleanup_multi_native", type: "function" },
    performStreamNative: { symbol: "jayess_curl_perform_stream_native", type: "function" }
  }
};
