// @jayess/openssl/index.js
import { randomBytesNative } from "./native/openssl.bind.js";
import { versionNative } from "./native/openssl.bind.js";
import { supportsHashNative } from "./native/openssl.bind.js";
import { supportsCipherNative } from "./native/openssl.bind.js";
import { hashNative } from "./native/openssl.bind.js";
import { hmacNative } from "./native/openssl.bind.js";
import { encryptNative } from "./native/openssl.bind.js";
import { decryptNative } from "./native/openssl.bind.js";
import { generateKeyPairNative } from "./native/openssl.bind.js";
import { publicEncryptNative } from "./native/openssl.bind.js";
import { privateDecryptNative } from "./native/openssl.bind.js";
import { signNative } from "./native/openssl.bind.js";
import { verifyNative } from "./native/openssl.bind.js";

export function randomBytes(length) {
  return randomBytesNative(length);
}

export function version() {
  return versionNative();
}

export function supportsHash(algorithm) {
  return supportsHashNative(algorithm);
}

export function supportsCipher(algorithm) {
  return supportsCipherNative(algorithm);
}

export function hash(algorithm, value) {
  return hashNative(algorithm, value);
}

export function hmac(algorithm, key, value) {
  return hmacNative(algorithm, key, value);
}

export function encrypt(options) {
  return encryptNative(options);
}

export function decrypt(options) {
  return decryptNative(options);
}

export function generateKeyPair(options) {
  return generateKeyPairNative(options);
}

export function publicEncrypt(options) {
  return publicEncryptNative(options);
}

export function privateDecrypt(options) {
  return privateDecryptNative(options);
}

export function sign(options) {
  return signNative(options);
}

export function verify(options) {
  return verifyNative(options);
}

export function tlsAvailable() {
  return tls.isAvailable();
}

export function tlsBackend() {
  return tls.backend();
}

export function tlsConnect(options) {
  return tls.connect(options);
}

export function tlsCreateServer(options, handler) {
  return tls.createServer(options, handler);
}
