// @jayess/libuv/native/libuv.bind.js
const f = () => {};
export const createLoopNative = f;
export const runNative = f;
export const runOnceNative = f;
export const stopNative = f;
export const closeLoopNative = f;
export const scheduleStopNative = f;
export const scheduleCallbackNative = f;
export const readFileNative = f;
export const watchSignalNative = f;
export const closeSignalWatcherNative = f;
export const watchPathNative = f;
export const closePathWatcherNative = f;
export const spawnProcessNative = f;
export const closeProcessNative = f;
export const createUDPNative = f;
export const bindUDPNative = f;
export const recvUDPNative = f;
export const sendUDPNative = f;
export const closeUDPNative = f;
export const createTCPServerNative = f;
export const listenTCPNative = f;
export const acceptTCPNative = f;
export const closeTCPServerNative = f;
export const createTCPClientNative = f;
export const connectTCPNative = f;
export const readTCPNative = f;
export const writeTCPNative = f;
export const closeTCPClientNative = f;
export const nowNative = f;

export default {
  sources: ["./libuv.c"],
  includeDirs: ["../../../../refs/libuv/include"],
  cflags: [],
  ldflags: [],
  platforms: {
    linux: {
      sources: [
        "../../../../refs/libuv/src/fs-poll.c",
        "../../../../refs/libuv/src/idna.c",
        "../../../../refs/libuv/src/inet.c",
        "../../../../refs/libuv/src/random.c",
        "../../../../refs/libuv/src/strscpy.c",
        "../../../../refs/libuv/src/strtok.c",
        "../../../../refs/libuv/src/thread-common.c",
        "../../../../refs/libuv/src/threadpool.c",
        "../../../../refs/libuv/src/timer.c",
        "../../../../refs/libuv/src/uv-common.c",
        "../../../../refs/libuv/src/uv-data-getter-setters.c",
        "../../../../refs/libuv/src/version.c",
        "../../../../refs/libuv/src/unix/async.c",
        "../../../../refs/libuv/src/unix/core.c",
        "../../../../refs/libuv/src/unix/dl.c",
        "../../../../refs/libuv/src/unix/fs.c",
        "../../../../refs/libuv/src/unix/getaddrinfo.c",
        "../../../../refs/libuv/src/unix/getnameinfo.c",
        "../../../../refs/libuv/src/unix/loop-watcher.c",
        "../../../../refs/libuv/src/unix/loop.c",
        "../../../../refs/libuv/src/unix/pipe.c",
        "../../../../refs/libuv/src/unix/poll.c",
        "../../../../refs/libuv/src/unix/process.c",
        "../../../../refs/libuv/src/unix/random-devurandom.c",
        "../../../../refs/libuv/src/unix/signal.c",
        "../../../../refs/libuv/src/unix/stream.c",
        "../../../../refs/libuv/src/unix/tcp.c",
        "../../../../refs/libuv/src/unix/thread.c",
        "../../../../refs/libuv/src/unix/tty.c",
        "../../../../refs/libuv/src/unix/udp.c",
        "../../../../refs/libuv/src/unix/linux.c",
        "../../../../refs/libuv/src/unix/procfs-exepath.c",
        "../../../../refs/libuv/src/unix/proctitle.c",
        "../../../../refs/libuv/src/unix/random-getrandom.c",
        "../../../../refs/libuv/src/unix/random-sysctl-linux.c"
      ],
      includeDirs: [
        "../../../../refs/libuv/src",
        "../../../../refs/libuv/src/unix"
      ],
      cflags: [
        "-D_GNU_SOURCE",
        "-D_POSIX_C_SOURCE=200112",
        "-pthread"
      ],
      ldflags: [
        "-pthread",
        "-ldl",
        "-lrt"
      ]
    }
  },
  exports: {
    createLoopNative: { symbol: "jayess_libuv_create_loop_native", type: "function" },
    runNative: { symbol: "jayess_libuv_run_native", type: "function" },
    runOnceNative: { symbol: "jayess_libuv_run_once_native", type: "function" },
    stopNative: { symbol: "jayess_libuv_stop_native", type: "function" },
    closeLoopNative: { symbol: "jayess_libuv_close_loop_native", type: "function" },
    scheduleStopNative: { symbol: "jayess_libuv_schedule_stop_native", type: "function" },
    scheduleCallbackNative: { symbol: "jayess_libuv_schedule_callback_native", type: "function" },
    readFileNative: { symbol: "jayess_libuv_read_file_native", type: "function" },
    watchSignalNative: { symbol: "jayess_libuv_watch_signal_native", type: "function" },
    closeSignalWatcherNative: { symbol: "jayess_libuv_close_signal_watcher_native", type: "function" },
    watchPathNative: { symbol: "jayess_libuv_watch_path_native", type: "function" },
    closePathWatcherNative: { symbol: "jayess_libuv_close_path_watcher_native", type: "function" },
    spawnProcessNative: { symbol: "jayess_libuv_spawn_process_native", type: "function" },
    closeProcessNative: { symbol: "jayess_libuv_close_process_native", type: "function" },
    createUDPNative: { symbol: "jayess_libuv_create_udp_native", type: "function" },
    bindUDPNative: { symbol: "jayess_libuv_bind_udp_native", type: "function" },
    recvUDPNative: { symbol: "jayess_libuv_recv_udp_native", type: "function" },
    sendUDPNative: { symbol: "jayess_libuv_send_udp_native", type: "function" },
    closeUDPNative: { symbol: "jayess_libuv_close_udp_native", type: "function" },
    createTCPServerNative: { symbol: "jayess_libuv_create_tcp_server_native", type: "function" },
    listenTCPNative: { symbol: "jayess_libuv_listen_tcp_native", type: "function" },
    acceptTCPNative: { symbol: "jayess_libuv_accept_tcp_native", type: "function" },
    closeTCPServerNative: { symbol: "jayess_libuv_close_tcp_server_native", type: "function" },
    createTCPClientNative: { symbol: "jayess_libuv_create_tcp_client_native", type: "function" },
    connectTCPNative: { symbol: "jayess_libuv_connect_tcp_native", type: "function" },
    readTCPNative: { symbol: "jayess_libuv_read_tcp_native", type: "function" },
    writeTCPNative: { symbol: "jayess_libuv_write_tcp_native", type: "function" },
    closeTCPClientNative: { symbol: "jayess_libuv_close_tcp_client_native", type: "function" },
    nowNative: { symbol: "jayess_libuv_now_native", type: "function" }
  }
};
